lib/debts/service.rb

```diff
@@ -6,26 +6,51 @@
 
 module Debts
   class Service < Common::Client::Base
+    attr_reader :file_number
+
     include Common::Client::Concerns::Monitoring
 
     configuration Debts::Configuration
 
     STATSD_KEY_PREFIX = 'api.debts'
 
-    def get_debts(body)
-      with_monitoring_and_error_handling do
-        GetDebtsResponse
-          .new(perform(:post, 'letterdetails/get', body).body)
-          .debts
-          .select do |debt|
-            debt['debtHistory'] = sort_by_date(debt['debtHistory'])
-            debt['payeeNumber'] == '00'
-          end
-      end
+    def initialize(user)
+      @user = user
+      @file_number = init_file_number
+      @debts = init_debts
+    end
+
+    def get_debts
+      {
+        has_dependent_debts: veteran_has_dependent_debts?,
+        debts: sorted_debts
+      }
+    end
+
+    def veteran_has_dependent_debts?
+      @debts.any? { |debt| debt['payeeNumber'] != '00' }
     end
 
     private
 
+    def init_file_number
+      bgs_file_number = BGS::PeopleService.new(@user).find_person_by_participant_id[:file_nbr]
+      bgs_file_number.presence || @user.ssn
+    end
+
+    def sorted_debts
```

Comment:
this is not a list of sorted debts, but debts with sorted histories.  I think this name is a little misleading and would probably do something like `debts_with_sorted_histories` myself

Reply:
Good call. Change has been made.